### DISCUSSION TITLE
Unable to search formula/casks

### DISCUSSION THREAD

**Discussion by andrew-wales** [Homebrew/discussions#4512] [Getting started] [answered]
### Output of `brew config`

```shell
HOMEBREW_VERSION: 4.0.18-57-g13d716b
ORIGIN: https://github.com/Homebrew/brew
HEAD: 13d716b08cb4248920e57c0a4c16c81f2ebc911f
Last commit: 3 hours ago
Core tap JSON: 17 May 11:45 UTC
HOMEBREW_PREFIX: /usr/local
HOMEBREW_CASK_OPTS: []
HOMEBREW_MAKE_JOBS: 12
Homebrew Ruby: 2.6.10 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: dodeca-core 64-bit kabylake
Clang: 14.0.3 build 1403
Git: 2.39.2 => /Library/Developer/CommandLineTools/usr/bin/git
Curl: 7.87.0 => /usr/bin/curl
macOS: 13.3.1-x86_64
CLT: 14.3.0.0.1.1679647830
Xcode: N/A
```

### Output of `brew doctor`

```shell
Your system is ready to brew.
```

### Description of issue

I'm trying to search for anything matching a given keyword, preferably in the description. `brew help desc` implies this is the command I need, but I suspect I'm misunderstanding something and would appreciate guidance. For example, searching for anything with the keyword "bash":

```
$ brew desc -s bash
Error: Calling brew desc is disabled! Use brew desc --eval-all or HOMEBREW_EVAL_ALL instead.
```

Following this rabbit hole:

```
$ brew desc -s --eval-all bash
==> Formulae

==> Casks
```

I thought `brew search` might help here, but no. I can search by formula/cask name, but not by description:

```
$ brew search --desc bash
Error: Calling brew search --desc is disabled! Use brew search --desc --eval-all or HOMEBREW_EVAL_ALL instead.
```

```
$ brew search --desc --eval-all bash
==> Formulae

==> Casks
```

Have I misunderstood a subtle context, or is my installation mangled in some way? Thanks in advance.

**Comment by Bo98** [comment]
Description searching is not particularly well supported at the moment (or anything that uses `--eval-all` really), but we're looking to solve that.

A workaround that will hopefully work for you is `HOMEBREW_EVAL_ALL=1 brew update`. This should update the description cache.

**Reply by andrew-wales** [reply]
I don't think this worked for me, unfortunately.

```
$ HOMEBREW_EVAL_ALL=1 brew update
==> Updated Homebrew from 13d716b08 to f7b322557.
No changes to formulae or casks.
$ brew desc -s --eval-all bash
==> Formulae

==> Casks
$ brew search --desc bash
==> Formulae

==> Casks
```

**Reply by CatGreen90** [reply, marked as answer]
As far as I can tell you have to have homebrew/core (and homebrew/cask for casks) tapped for it to work correctly. ``brew uses`` is also broken with untapped homebrew/core.

**Reply by Bo98** [reply]
> I don't think this worked for me, unfortunately.
> 
> ```
> $ HOMEBREW_EVAL_ALL=1 brew update
> ==> Updated Homebrew from 13d716b08 to f7b322557.
> No changes to formulae or casks.
> $ brew desc -s --eval-all bash
> ==> Formulae
> 
> ==> Casks
> $ brew search --desc bash
> ==> Formulae
> 
> ==> Casks
> ```

Unfortunately it won't work if it outputs `No changes to formulae or casks.`.

But yes, I forgot to mention it also requires a tapped Homebrew/core or Homebrew/cask currently, but the first time you tap them it will update the cache if you have `HOMEBREW_EVAL_ALL=1` set while doing so.

**Reply by andrew-wales** [reply]
This was the winning combination. Thanks, all!

```
$ export HOMEBREW_EVAL_ALL=1
$ brew tap homebrew/core
[...]
$ brew tap homebrew/cask
[...]
$ brew desc -s bash | wc -l
      64
$
```
